src/components/editor/Bindings/Backfill/useUpdateBackfillCounter.ts

```diff
@@ -0,0 +1,135 @@
+import { useResourceConfig_backfilledCollections } from 'stores/ResourceConfig/hooks';
+import { Schema } from 'types';
+import { hasLength } from 'utils/misc-utils';
+import { getBackfillCounter, getCollectionName } from 'utils/workflow-utils';
+import {
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_mutate,
+} from '../../Store/hooks';
+
+export interface BindingMetadata {
+    bindingIndex: number;
+    collection: string;
+}
+
+const evaluateBackfillCounter = (
+    binding: Schema,
+    increment: BooleanString
+): Schema => {
+    let counter = getBackfillCounter(binding);
+
+    if (increment === 'true') {
+        counter = counter + 1;
+    } else if (counter > 0) {
+        counter = counter - 1;
+    }
+
+    binding.backfill = counter;
+
+    return binding;
+};
+
+function useUpdateBackfillCounter() {
+    const intl = useIntl();
+    const entityType = useEntityType();
+
+    // Draft Editor Store
+    const draftId = useEditorStore_persistedDraftId();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    // Resource Config Store
+    const backfilledCollections = useResourceConfig_backfilledCollections();
+
+    const updateBackfillCounter = useCallback(
+        async (
+            draftSpec: DraftSpecQuery,
+            increment: BooleanString,
+            bindingMetadata: BindingMetadata[]
+        ) => {
+            const bindingMetadataExists = hasLength(bindingMetadata);
+
+            const invalidBindingIndex = bindingMetadataExists
+                ? bindingMetadata.findIndex(
+                      ({ bindingIndex }) => bindingIndex === -1
+                  )
+                : -1;
+
+            if (!mutateDraftSpecs || invalidBindingIndex > -1) {
+                const errorMessageId = bindingMetadataExists
+                    ? 'workflows.collectionSelector.manualBackfill.error.message.singleCollection'
+                    : 'workflows.collectionSelector.manualBackfill.error.message.allBindings';
+
+                const errorMessageValues = bindingMetadataExists
+                    ? {
+                          collection:
+                              bindingMetadata[invalidBindingIndex].collection,
+                      }
+                    : undefined;
+
+                return Promise.reject({
+                    ...BASE_ERROR,
+                    message: intl.formatMessage(
+                        { id: errorMessageId },
+                        errorMessageValues
+                    ),
+                });
+            }
+
+            const spec: Schema = draftSpec.spec;
+
+            if (bindingMetadataExists) {
+                bindingMetadata.forEach(({ bindingIndex }) => {
+                    if (bindingIndex > -1) {
+                        spec.bindings[bindingIndex] = evaluateBackfillCounter(
+                            spec.bindings[bindingIndex],
+                            increment
+                        );
```

Comment:
This and the call below should not be setting the entire binding. This should just update the specific property we're controlling.

Reply:
This is mildly a case of logical semantics, but agreed.